### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by ishimwepati**
Hello !

In this branch, I've implemented several enhancements:

1. Ensured that user names cannot be blank, and comments counters must be non-negative integers or zero.
2. Implemented validations for posts, requiring titles not to be blank, limiting them to a maximum of 250 characters. Additionally, the counters for comments and likes in posts must be non-negative numbers or zero.

Furthermore, for each validation introduced, a set of 18 tests has been meticulously crafted and successfully executed, confirming the robustness of the implementation. All tests have passed without any issues.

Your suggestions are most welcome !

### CODE REVIEW

File: spec/models/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  it 'post_counter should be integer' do
+    subject.post_counter = 'a'
+    expect(subject).to_not be_valid
+  end
+
+  it 'post_counter should be greater than or equal to zero' do
+    subject.post_counter = -1
+    expect(subject).to_not be_valid
+  end
```
Comment (tufoinnkuo10) [inline spec/models/users_spec.rb:26]: - [x] As stated by the guidelines, kindly add unit specs for all of your models' methods and validations.... Please add test for the user model method, ...` most_recent_posts` please ensure your test are working well. Nice work.

File: spec/models/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  it 'likes_counter should be integer' do
+    subject.likes_counter = 'c'
+    expect(subject).to_not be_valid
+  end
+
+  it 'likes_counter should be greater than or equal to zero' do
+    subject.likes_counter = -2
+    expect(subject).to_not be_valid
+  end
```
Comment (tufoinnkuo10) [inline spec/models/posts_spec.rb:46]: - [x] wonderful job. Please, like stated in the guidelines, any custom methods in the models should have a corresponding unit test. Kindly add unit test for the post model methods of;;
---- Most recent comments and updating the post counter. Please kindly ensure your test are working fine.

- [x] please kindly add test for the methods in the comments and like model as well.. like the update comments and like counter methods. kindly ensure your test are passing. nice work.

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @ishimwepati 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Gitflow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally

Required Changes ♻️
Check the comments under the review
Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: spec/models/users_spec.rb (hunk 1 above)
Comment (Davidosky007) [reply spec/models/users_spec.rb:26]: * Good job working on the requested changes but i noticed some errors while running your tests below is a screenshot of the said error Kindly fix them. 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @ishimwepati 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* Well-written PR 👍 
* Good looking readme 👍 
* No linter errors 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @ishimwepati ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- All tests are passing ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**